app/shop.py

```diff
@@ -0,0 +1,41 @@
+import datetime
+
+
+class Shop:
+    def __init__(
+            self,
+            name: str,
+            location: list[int],
+            products: dict
+    ) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+        self.each_product_cost = {}
+        self.total_cost = 0
+
+    def purchase(self, product_cart: dict) -> int:
+        for product in product_cart:
+            try:
+                product_cost = product_cart[product] * self.products[product]
+                if (isinstance(product_cost, float)
+                        and product_cost.is_integer()):
+                    product_cost = int(product_cost)
+                self.each_product_cost[product] = product_cost
+                self.total_cost += product_cost
+            except KeyError:
+                print("There is no such product at store!")

```

Comment:
Using `print` statements for error handling is not recommended for production code. Consider using logging or raising an exception instead.